cmd/api.go

```diff
@@ -163,23 +164,21 @@ func (o *apiOptions) validate(c *config.Config) error {
 
 	// In case we want to scaffold a resource API we need to do some checks
 	if o.doResource {
-		// Skip the following check for v1 as resources aren't tracked
-		if !c.IsV1() {
-			// Check that resource doesn't exist or flag force was set
-			if !o.force {
-				resourceExists := false
-				for _, r := range c.Resources {
-					if r.Group == o.resource.Group &&
-						r.Version == o.resource.Version &&
-						r.Kind == o.resource.Kind {
-						resourceExists = true
-						break
-					}
-				}
-				if resourceExists {
-					return errors.New("API resource already exists")
+
+		// Check that resource doesn't exist or flag force was set
+		if !o.force {
+			resourceExists := false
+			for _, r := range c.Resources {
+				if r.Group == o.resource.Group &&
+					r.Version == o.resource.Version &&
+					r.Kind == o.resource.Kind {
+					resourceExists = true
+					break
 				}
 			}
+			if resourceExists {
```

Comment:
```suggestion
			if c.HasResource(o.resource.GVK()) {
```

Config already has a function to check if the resource is tracked (== exists): https://github.com/kubernetes-sigs/kubebuilder/blob/f0c25d083cc896c3b3328b8ecb9108a6dd7ed9d8/pkg/model/config/config.go#L71-L86

Reply:
It should be changed in the master. Other PR. 
(not part of the scope)

Reply:
I'll submit a PR for this.